libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -685,7 +555,7 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
         // Handling remotely deleted records
         if (err.code == 404) {
             if (mergeMode == SFSyncStateMergeModeOverwrite) {
-                [target createOnServer:objectType fields:fields completionBlock:completeBlockCreate failBlock:failBlock];
+                [target createOnServer:self record:record fieldlist:sync.options.fieldlist completionBlock:completeBlockCreate failBlock:failBlock];
```

Comment:
Create/update field list did not exist in 5.0 (they are new in 5.1).
First I added them to the sync options, but I think they make more sense as an attribute of the sync target. (Sync options are for things that make sense / are processed by the sync manager).